src/app/Fake.Azure.Emulators/Emulators.fs

```diff
@@ -0,0 +1,85 @@
+﻿/// Contains tasks to control the local Azure Emulator
+module Fake.Azure.Emulators
+
+open System
+open Fake.Core
+open Fake.IO
+open Fake.IO.FileSystemOperators
+
+/// A type for the controlling parameter
+[<CLIMutable>]
+type private AzureEmulatorParams = {
+    StorageEmulatorToolPath:Lazy<string>
+    CSRunToolPath:string
+    TimeOut:TimeSpan
+    }
+
+/// Base path for getting tools from Microsoft SDKs
+let msSdkBasePath = Environment.ProgramFilesX86 @@ "Microsoft SDKs"
+
+/// The default parameters for Azure emulators
+let private AzureEmulatorDefaults = {
+    StorageEmulatorToolPath =
+        lazy
+            let path = msSdkBasePath @@ @"\Azure\Storage Emulator\AzureStorageEmulator.exe"
+            if File.exists path then path
+            else failwith (sprintf "Unable to locate Azure Storage Emulator at %s" path)
+    CSRunToolPath = "\"C:\Program Files\Microsoft SDKs\Windows Azure\Emulator\csrun.exe\""
+    TimeOut = TimeSpan.FromMinutes 5.
+    }
+
+let private (|StorageAlreadyStarted|StorageAlreadyStopped|Ok|OtherError|) = function
+    | 0 -> Ok
+    | -5 -> StorageAlreadyStarted
+    | -6 -> StorageAlreadyStopped
+    | _ -> OtherError
+
+/// Stops the storage emulator
+let StopStorageEmulator = (fun _ ->
+    match Process.Exec (fun info ->
+        { info with
+            FileName = AzureEmulatorDefaults.StorageEmulatorToolPath.Value
+            Arguments = "stop" }) AzureEmulatorDefaults.TimeOut with
+    | Ok | StorageAlreadyStopped -> ()
+    | _ -> failwithf "Azure Emulator Failure on stop Storage Emulator"
```

Comment:
Maybe we need some more information here, if you want to hide it you can use inner exceptions, like
`raise <| exn("Azure Emulator Failure ...", exn(sprintf "Process exited with code %d" exitCode))`

But that is only a suggestion if you left that information out on purpose, otherwise we might just include the exit code in the error message...

Reply:
I copied all this from the `FakeLib` project.